File: problem.3/problem3.pl
#!/usr/bin/perl
use strict;

use lib '../lib';
use Euler::Prime::Modulus 'is_prime';

sub prime_factor {
	my $x = shift;

	my $best ;

	my ($max, $test) = ( $x, 3 );

	while ( $max >= $test){
		if ( is_prime( $test ) && ( $max % $test == 0)){
			$best = $test;
			$max = $max / $test;
		} else {
			$test = $test + 2;
		}
	}
	return $best;
}

print prime_factor( 600851475143 ) . "\n";

File: problem.3/problem3.py
#!python3

x = 600851475143
n = 2
fp = set()

while x != n :
    if x % n == 0 :
        x = x / n
        fp.add(n)
        n = 2
    else :
        n += 1

fp.add(n)
print(max(fp))
